src/state/themes-mixin.ts

```diff
@@ -89,6 +85,7 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
       }
 
       themeSettings = { ...this.hass.selectedTheme, dark: darkMode };
+      this.hass.themes.theme = themeName;
```

Comment:
This will not trigger an update of `themes`:
```suggestion
      this._updateHass({
          themes: { ...this.hass.themes!, theme: themeName },
      });```